lib/Scheduler/bb_spill.cpp

```diff
@@ -847,8 +855,101 @@ FUNC_RESULT BBWithSpill::Enumerate_(Milliseconds startTime,
 }
 /*****************************************************************************/
 
-InstCount BBWithSpill::UpdtOptmlSched(InstSchedule *crntSched,
-                                      LengthCostEnumerator *) {
+SmallVector<InstCount, 4>
+BBWithSpill::UpdtOptmlSched(InstSchedule *crntSched,
+                            LengthCostEnumerator *node) {
+
+  if (isTwoPassEnabled()) {
+    if (!IsSecondPass())
+      return UpdtOptmlSchedFrstPss(crntSched, node);
+    else
+      return UpdtOptmlSchedScndPss(crntSched, node);
+  }
+
+  else
+    return UpdtOptmlSchedWghtd(crntSched, node);
+}
+
+/*****************************************************************************/
+
+SmallVector<InstCount, 4>
+BBWithSpill::UpdtOptmlSchedFrstPss(InstSchedule *crntSched,
+                                   LengthCostEnumerator *) {
+  InstCount crntCost;
+  InstCount crntExecCost;
+  InstCount TmpSpillCost;
```

Comment:
Move these to the definition. Having variable declarations at the top of a function is an old C-ism (for old versions of C) which makes code harder to read.

Reply:
Although, if this is just existing code, it's okay to leave as-is. I'm not sure because I think this is copied code. It's safe to move them to the definition, though, and I prefer to do that when I touch the existing code.